src/toil/leader.py

```diff
@@ -50,15 +52,54 @@ def __init__(self, jobStore):
                               args=(jobStore, self._stop))
         self._worker.start()
 
-    @staticmethod
-    def statsAndLoggingAggregator(jobStore, stop):
+    @classmethod
+    def writeLogFiles(cls, jobNames, jobLogList, config):
+        def createName(logPath, jobName, logExtension):
+            logName = jobName.replace('-', '--')
+            logName = logName.replace('/', '-')
+            logName = logName.replace(' ', '_')
+            logName = logName.replace("'", '')
+            counter = 0
+            while True:
+                suffix = str(counter).zfill(3) + logExtension
+                fullName = os.path.join(logPath, logName + suffix)
+                if not os.path.exists(fullName):
+                    return fullName
+                counter += 1
+
+        mainFileName = jobNames[0]
+        path = None
+        writeFn = None
+        extension = '.log'
+        if config.writeLogs:
+            path = config.writeLogs
+            writeFn = open
+        if config.writeLogsGzip:
+            path = config.writeLogsGzip
+            writeFn = gzip.open
+            extension += '.gz'
+        if path is None:
+            # we don't have anywhere to write the logs, return now
+            return
+        fullName = createName(path, mainFileName, extension)
+        with writeFn(fullName, 'w') as f:
+            f.writelines(l + '\n' for l in jobLogList)
+        for alternateName in jobNames[1:]:
+            # There are chained jobs in this output - indicate this with a symlink
+            # of the job's name to this file
+            name = createName(path, alternateName, extension)
+            os.symlink(fullName, name)
```

Comment:
The symlink should be relative. Absolute symlinks would break if the user moved the directory around, or tarred it and then untarred to a different location.